File: spelling_bee.py
import re
from collections import Counter

words = []
with open("words_alpha.txt", encoding="utf-8") as f:
    for l in f:
        if len(l) > 3:
            if "\n" in l:
                words.append(l[:-1])
            else:
                words.append(l)

def spelling_bee(letters, center):
    result = []
    letters = letters.lower()
    center = center.lower()
    regex = r"^([" + letters + "])*" + center + "+" + "([" + letters + "])*$"
    for w in words:
        if re.match(regex, w) != None and len(w) > 3:
            result.append(w)
    result_set = set(result)
    res2 = sorted(result_set)

    for w in res2:
        print(w.upper())
    print(f"Found {len(result)} words!")

    for w in res2:
        c = Counter(w)
        if len(c) >= 7:
            print(f"Possible Panagaram: {w.upper()}")


spelling_bee("leiacm", "f")

File: src/spellingBee.js
const fs = require("fs");
const text = fs.readFileSync("words_alpha.txt", "utf-8");
const words = text.split("\r\n");

function spellingBee(letters, center) {
  const result = [];
  letters = letters.toLowerCase();
  center = center.toLowerCase();
  const re = new RegExp(
    `^([${letters}])*${center}+([${letters}])*$`
  );
  // gather valid words
  for (let w of words) {
    if (w.match(re) !== null && w.length > 3) {
      result.push(w);
    }
  }
  // print all valid words
  result.sort();
  for (let g of result) {
    console.log(g.toUpperCase());
  }
  console.log(`Found ${result.length} words!`);

  // print possible panagrams
  for (let p of result) {
    if (p.length >= 7) {
      let p_set = new Set(p);
      if (p_set.size >= 7) {
        console.log(`Possible Panagram: ${p.toUpperCase()}`);
      }
    }
  }
}

spellingBee("leiacm", "f");
